### PR TITLE
Update Google Sign-In plugin for new plugin APIs

### PR DESCRIPTION
**Pull request by collinjackson**
Fixes flutter/flutter#8581

### CODE REVIEW

File: example/ios/Runner/Info.plist [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	<key>CFBundleInfoDictionaryVersion</key>
 	<string>6.0</string>
 	<key>CFBundleName</key>
-	<string>example</string>
+	<string>Google Sign-In Example</string>
```
Comment (jakobr-google) [inline example/ios/Runner/Info.plist]: Nit: According to the [specification](https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/TP40009249-109585): `CFBundleName` (String - iOS, macOS) specifies the short name of the bundle. This name should be less than 16 characters long.

You might want `CFBundleDisplayName`, which is what's shown on the home screen, etc. And falls back to CFBundleName if it isn't defined.

File: LICENSE [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
-LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// Copyright 2017 Your Company. All rights reserved.
```
Comment (jakobr-google) [inline LICENSE]: Might want to substitute 'Your Company' in this file. Or keep the old LICENSE file.

File: android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,38 @@
+group 'io.flutter.plugin'
```
Comment (jakobr-google) [inline android/build.gradle]: io.flutter.googlesignin?

File: android/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+  package="com.yourcompany.googlesignin"
```
Comment (jakobr-google) [inline android/src/main/AndroidManifest.xml]: io.flutter.googlesignin?

File: example/.flutter-plugins [hunk 5]
```diff
@@ -0,0 +1 @@
+google_sign_in=../
```
Comment (jakobr-google) [inline example/.flutter-plugins]: Nit: This file will be re-generated on every build, so you don't have to check it in. Doesn't hurt either, though.

It's in .gitignore in the new project template.

File: ios/Classes/GoogleSignInPlugin.m [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    FlutterMethodChannel *channel = [FlutterMethodChannel
+        methodChannelNamed:@"plugins.flutter.io/google_sign_in"
+           binaryMessenger:flutterView
+                     codec:[FlutterStandardMethodCodec sharedInstance]];
+    _accountRequests = [[NSMutableArray alloc] init];
+    [GIDSignIn sharedInstance].delegate = self;
+    [GIDSignIn sharedInstance].uiDelegate = flutterView;
+    [channel setMethodCallHandler:^(FlutterMethodCall *call,
+                                    FlutterResultReceiver result) {
+      [self handleMethodCall:call result:result];
```
Comment (jakobr-google) [inline ios/Classes/GoogleSignInPlugin.m]: @mravn-google I'm starting to suspect this'll be a common pattern. We should consider if a protocol-based approach would work better.

File: ios/Classes/GoogleSignInPlugin.m [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    }];
+  }
+  return self;
+}
+
+- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResultReceiver)result {
+    if ([call.method isEqualToString:@"init"]) {
+        NSError *error;
+        [[GGLContext sharedInstance] configureWithError:&error];
```
Comment (collinjackson) [inline ios/Classes/GoogleSignInPlugin.m]: I should check for errors here

**Comment by collinjackson** [comment]
thanks for the feedback, somehow I accidentally merged it but I'll address review feedback now

**Pull request merged**